File: ios/Dish.swift
//
//  Dish.swift
//  GourmetSB
//
//  Created by siyaoli on 2/24/18.
//  Copyright © 2018 siyaoli. All rights reserved.
//

import Foundation

public class Dish {
    //basic attributes
    private let dishName : String;
    private let fromDC : Int;
    private let section : String;
    private let mealTime : String;
    private let foodType : Int;
    //private boolean isNuts;
//    private var score : Double = 1;
    
    //attributes for better sort
    
    init(dishName : String, fromDC : Int, section : String, mealTime : String, foodType : Int) {
        self.dishName = dishName;
        self.fromDC = fromDC;
        self.section = section;
        self.mealTime = mealTime;
        self.foodType = foodType;
    }
    
    public func getDishName() -> String{
        return dishName;
    }
    
    public func getFoodType() -> Int {
        return foodType;
    }
    
    public func getFromDC() -> Int {
        return fromDC;
    }
    
    public func getSection() -> String {
        return section;
    }
    
//    public func getScore() -> Double {
//        return score;
//    }
//    
//    public func setScore(score : Double) {
//        self.score = score;
//    }
    
//    public void setFoodType(int foodType) {
//    this.foodType = foodType;
//    }
    
//    public boolean isNuts() {
//    return isNuts;
//    }
    
//    public void setNuts(boolean nuts) {
//    isNuts = nuts;
//    }
    
    public func getMealTime() -> String {
        return mealTime;
    }
}

File: AllSourceCodesForAndroid/Dish.java
package com.example.shihualu.gourmettest.Module;

public class Dish {
    //basic attributes
    private String dishName;
    private int fromDC;
    private String section;
    private int mealTime;
    private int foodType;
    private boolean isNuts;
    private double score=2;

    //attributes for better sort

    public Dish(String dishName, int fromDC, String section, int mealTime) {
        this.dishName = dishName;
        this.fromDC = fromDC;
        this.section = section;
        this.mealTime = mealTime;
    }



    public String getDishName() {
        return dishName;
    }

    public int getFoodType() {
        return foodType;
    }

    public int getFromDC() {
        return fromDC;
    }

    public String getSection() {
        return section;
    }

    public double getScore() {
        return score;
    }

    public void setScore(double score) {
        this.score = score;
    }

    public void setDishName(String dishName) {
        this.dishName = dishName;
    }

    public void setFoodType(int foodType) {
        this.foodType = foodType;
    }

    public void setFromDC(int fromDC) {
        this.fromDC = fromDC;
    }

    public void setSection(String section) {
        this.section = section;
    }

    public boolean isNuts() {
        return isNuts;
    }

    public void setNuts(boolean nuts) {
        isNuts = nuts;
    }

    public int getMealTime() {
        return mealTime;
    }

    public void setMealTime(int mealTime) {
        this.mealTime = mealTime;
    }
}